src/protocol/codecs/topic.rs

```diff
@@ -40,6 +44,27 @@ impl MsgOfInterest {
     }
 }
 
+impl MsgOfInterest {
+    /// Convert the message to a corresopnding [Topic].
+    #[allow(clippy::wrong_self_convention)]
+    fn to_topic(self) -> Topic {
+        let mut value = String::new();
+
+        if !self.tags.is_empty() {
+            for tag in self.tags {
+                value.push_str(&tag.to_string());
+                value.push(',');
+            }
+            value.pop().unwrap(); // Remove a trailing ','.
```

Comment:
I think `join`, `map` and `collect` will do the same in more idiomatic way.

Reply:
aha! let me try it 💪

Reply:
It looks nicer! Thanks!
Done